@stellar/design-system/src/components/TextLink/index.tsx

```diff
@@ -1,22 +1,81 @@
 import React from "react";
+import { ButtonIcon } from "../utils/ButtonIcon";
 import "./styles.scss";
 
-export enum TextLinkVariant {
+enum TextLinkVariant {
   primary = "primary",
   secondary = "secondary",
 }
 
+interface TextLinkComponent {
+  variant: typeof TextLinkVariant;
+}
+
 interface TextLinkProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
-  children: string;
+  iconLeft?: React.ReactNode;
+  iconRight?: React.ReactNode;
   variant?: TextLinkVariant;
+  disabled?: Boolean;
+  underline?: Boolean;
+  children: string | React.ReactNode;
 }
 
-export const TextLink: React.FC<TextLinkProps> = ({
+export const TextLink: React.FC<TextLinkProps> & TextLinkComponent = ({
+  iconLeft,
+  iconRight,
   variant = TextLinkVariant.primary,
+  disabled,
+  underline,
   children,
   ...props
-}) => (
-  <a className="TextLink" data-variant={variant} {...props}>
-    {children}
-  </a>
-);
+}) => {
+  const { href, onClick } = props;
+  const isExternalLink = href?.startsWith("http") || href?.startsWith("//");
```

Comment:
Hm, I'm not sure I understand why an href starting with `http` or `//` definitely means a link should open in a new tab

Reply:
In any case, I think we might be better suited making `isExternalLink` a prop so a user has more control over what opens in a new tab rather than trying to guess what they want